packages/system/Accounts/plugin/accounts/src/interfaces/admin.rs

```diff
@@ -34,31 +30,25 @@ fn assert_valid_account(account: &str) {
     assert!(account_details.is_some(), "Invalid account name");
 }
 
-#[derive(Serialize, Debug)]
-struct SetAuthCookieReqPayload {
-    accessToken: String,
-}
-
 impl Admin for AccountsPlugin {
     fn login_direct(app: String, user: String) {
         assert_caller_admin("login_direct");
 
         assert_valid_account(&user);
 
-        let query_token = TransactAuthApi::get_query_token(&Client::get_receiver(), &user).unwrap();
-
-        let payload = SetAuthCookieReqPayload {
-            accessToken: query_token.to_string(),
-        };
-
-        let req: PostRequest = PostRequest {
-            endpoint: String::from("/common/set-auth-cookie"),
-            body: BodyTypes::Json(serde_json::to_string(&payload).unwrap()),
-        };
-        HostAdmin::post(&app, &req).unwrap();
+        // We want to authenticate as the user who is logging in to the 3rd-party app.
+        // So we need to log in as this user in accounts because accounts is the active app
+        // (because we redirected to accounts to do the login).
+        AppsTable::new(&Client::get_receiver()).login(&user);
 
         AppsTable::new(&app).login(&user);
         UserTable::new(&user).add_connected_app(&app);
+
+        // TODO: if set_query_token() fails, logout
+        HostAuth::set_logged_in_user(&user, &app);
```

Comment:
I don't believe this is necessary anymore @James-Mart but please do correct me if I'm missing a detail.